stateMutators.js

```diff
@@ -0,0 +1,111 @@
+const gameState = {
+    basketPos: 50,
+    basketValue: 0,
+    score: 0,
+    livesRemaining: 3,
+    gameLevel: 1,
+    gameMode: 'running',
+    fallingObjects: [],
+}
+
+const setBasketPos = (state, value) => ({
+    ...state,
+    basketPos: value,
+});
+
+const moveBasketLeftKey = (state, value) => ({
+    ...state,
+    basketPos: basketPos - value,
+});
+
+const moveBasketRightKey = (state, value) => ({
+    ...state,
+    basketPos: basketPos + value,
+});
+
+const moveBasketMouse = (state, value) => ({
+    ...state,
+    basketPos: value,
+});
+
+const setBasketValue = (state, value) => ({
+    ...state,
+    basketValue: value,
+});
+
+const resetBasketValue = (state) => ({
+    ...state,
+    basketValue: 0,
+});
+
+const setScore = (state, value) => ({
+    ...state,
+    score: value,
+});
+
+const resetScore = (state) => ({
+    ...state,
+    score: 0,
+});
+
+const setLivesRemaining = (state, value) => ({
+    ...state,
+    livesRemaining: value,
+});
+
+const resetLivesRemaining = (state) => ({
+    ...state,
+    livesRemaining: 3,
+});
+
+const setGameLevel = (state, value) => ({
+    ...state,
+    gameLevel: value,
+});
+
+const resetGameLevel = (state) => ({
+    ...state,
+    gameLevel: 0,
+});
+
+const setGameMode = (state, value) => ({
+    ...state,
+    gameMode: value,
+});
+
+function addFallingObject (state, fallingObj) {
```

Comment:
This and the mutators below are a little bit different to the others in that they don't return anything, while the previous all return the new state. i.e., their [function/type signature](https://developer.mozilla.org/en-US/docs/Glossary/Signature/Function) is different

These ones should be updated to have the same signature so they can be used in the same ways

Reply:
updated